deployment/ccip/changeset/cs_sync_usdc_domains_with_chains.go

```diff
@@ -0,0 +1,165 @@
+// As such, it is expected that ConfigureTokenPoolContractsChangeset is executed before running this changeset.
+func SyncUSDCDomainsWithChainsChangeset(env deployment.Environment, c SyncUSDCDomainsWithChainsConfig) (deployment.ChangesetOutput, error) {
+	state, err := LoadOnchainState(env)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+	if err := c.Validate(env, state); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid SyncUSDCDomainsWithChainsConfig: %w", err)
+	}
+	readOpts := &bind.CallOpts{Context: env.GetContext()}
+
+	deployerGroup := NewDeployerGroup(env, state, c.MCMS)
+
+	for chainSelector, usdcChainConfig := range c.USDCConfigsByChain {
+		chain := env.Chains[chainSelector]
+		chainState := state.Chains[chainSelector]
+		writeOpts, err := deployerGroup.GetDeployer(chainSelector)
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to get transaction opts for %s", chain)
+		}
+
+		usdcTokenPool := chainState.USDCTokenPools[usdcChainConfig.Version]
+		supportedChains, err := usdcTokenPool.GetSupportedChains(readOpts)
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to fetch supported chains from USDC token pool with address %s on %s: %w", usdcTokenPool.Address(), chain, err)
+		}
+
+		domainUpdates := make([]usdc_token_pool.USDCTokenPoolDomainUpdate, 0)
+		for _, remoteChainSelector := range supportedChains {
+			remoteChainState := state.Chains[remoteChainSelector]
+			remoteUSDCChainConfig := c.USDCConfigsByChain[remoteChainSelector]
+			remoteUSDCTokenPool := remoteChainState.USDCTokenPools[remoteUSDCChainConfig.Version]
+
+			var desiredAllowedCaller [32]byte
+			remoteUSDCTokenPoolAddressBytes := remoteUSDCTokenPool.Address().Bytes()
+			for i, j := len(desiredAllowedCaller)-len(remoteUSDCTokenPoolAddressBytes), 0; i < len(desiredAllowedCaller); i, j = i+1, j+1 {
+				desiredAllowedCaller[i] = remoteUSDCTokenPoolAddressBytes[j]
+			}
+			desiredDomainIdentifier := c.ChainSelectorToUSDCDomain[remoteChainSelector]
+
+			currentDomain, err := usdcTokenPool.GetDomain(readOpts, remoteChainSelector)
+			if err != nil {
+				return deployment.ChangesetOutput{}, fmt.Errorf("failed to fetch domain for %d from USDC token pool with address %s on %s: %w", remoteChainSelector, usdcTokenPool.Address(), chain, err)
+			}
+			// If any parameters are different, we need to add a setDomains call
+			if currentDomain.AllowedCaller != desiredAllowedCaller ||
+				currentDomain.DomainIdentifier != desiredDomainIdentifier {
+				domainUpdates = append(domainUpdates, usdc_token_pool.USDCTokenPoolDomainUpdate{
+					AllowedCaller:     desiredAllowedCaller,
+					Enabled:           true,
+					DomainIdentifier:  desiredDomainIdentifier,
+					DestChainSelector: remoteChainSelector,
+				})
+			}
+		}
+
+		if len(domainUpdates) > 0 {
+			_, err := usdcTokenPool.SetDomains(writeOpts, domainUpdates)
```

Comment:
validating token pool ownership anywhere?

Reply:
Yea that check is present in SyncUSDCDomainsWithChainsConfig.Validate